tests/aws/services/s3/test_s3.py

```diff
@@ -4251,20 +4256,26 @@ def test_create_bucket_head_bucket(self, snapshot, aws_client):
                 snapshot.transform.regex(r"s3\.amazonaws\.com", "<host>"),
                 snapshot.transform.regex(r"s3\.localhost\.localstack\.cloud:4566", "<host>"),
                 snapshot.transform.regex(r"s3\.localhost\.localstack\.cloud:443", "<host>"),
+                snapshot.transform.key_value("x-amz-bucket-region"),
             ]
         )
 
         try:
-            response = aws_client.s3.create_bucket(Bucket=bucket_1)
+            client = aws_client_factory(region_name="us-east-1").s3
+            response = client.create_bucket(Bucket=bucket_1)
             snapshot.match("create_bucket", response)
 
             response = aws_client.s3.create_bucket(
                 Bucket=bucket_2,
-                CreateBucketConfiguration={"LocationConstraint": "us-west-1"},
+                CreateBucketConfiguration={
+                    "LocationConstraint": "us-east-2"
+                    if TEST_AWS_REGION_NAME != "us-east-2"
+                    else "us-west-2"
+                },
```

Comment:
I'm a bit confused by this bit: is the intention to use a different region than `TEST_AWS_REGION_NAME`? could we use `SECONDARY_TEST_AWS_REGION_NAME` since it will never be same as `TEST_AWS_REGION_NAME`?

Reply:
I suppose that's good as long as the second region is not us-east-1

Reply:
@viren-nadkarni I have made the required changes.

Reply:
@viren-nadkarni thanks for pointing this out, I have made the required changes.

Reply:
@sannya-singal would this be applicable at other places like [here](https://github.com/localstack/localstack/pull/9560/files#diff-9e996d8db48c7a9aab34a4fb63fdfc3e2e5ddc9da017632edc42407e121ac592R4813-R4815) or [here](https://github.com/localstack/localstack/pull/9560/files#diff-9e996d8db48c7a9aab34a4fb63fdfc3e2e5ddc9da017632edc42407e121ac592R4881-R4882)?

Reply:
This test is specifically comparing `Arn` here. We need to either write a transformer for reference replacement of the region or create region specific clients, the above implementation is just a workaround for now.

Reply:
@viren-nadkarni I have made the required changes here- https://github.com/localstack/localstack/pull/9570